packer/builder/azure/smapi/step_create_image.go

```diff
@@ -88,7 +88,8 @@ func (s *StepCreateImage) Cleanup(state multistep.StateBag) {
 				return vmdisk.NewClient(client).DeleteDisk(diskName, true)
 			}, retry.ConstantBackoffRule("busy", func(err management.AzureError) bool {
 				return strings.Contains(err.Message, "is currently performing an operation on deployment") ||
-					strings.Contains(err.Message, "is currently in use by virtual machine")
+					strings.Contains(err.Message, "is currently in use by virtual machine") ||
+					strings.Contains(err.Message, "connection reset by peer")
```

Comment:
This won't work, since this function is only called for AzureErrors,.. ie this is only for 'functional' retries if that makes sense... :-)
If we want to do it in this code base, we should do it either here: https://github.com/Azure/packer-azure/blob/master/packer/builder/azure/smapi/retry/operations.go#L33
Or, alternatively, we should wrap the Azure client that we create in builder.go